conans/test/unittests/test_client/test_graph/test_range_resolver.py

```diff
@@ -0,0 +1,39 @@
+import unittest
+from conans.client.graph.range_resolver import _parse_versionexpr
+from conans.test.utils.tools import TestBufferConanOutput
+
+
+class ParseVersionExpr(unittest.TestCase):
+    def test_backwards_compatibility(self):
+        output = TestBufferConanOutput()
+        self.assertEqual(_parse_versionexpr("2.3, 3.2", output), ("2.3 3.2", True, False))
+        self.assertEqual(_parse_versionexpr("2.3, <=3.2", output), ("2.3 <=3.2", True, False))
+
+    def test_only_loose(self):
+        output = TestBufferConanOutput()
+        self.assertEqual(_parse_versionexpr("2.3 ,3.2, loose=True", output), ("2.3 3.2", True, False))
+        self.assertEqual(_parse_versionexpr("2.3 3.2, loose=False", output), ("2.3 3.2", False, False))
+        self.assertEqual(_parse_versionexpr("2.3 3.2, loose  = False", output), ("2.3 3.2", False, False))
+        self.assertEqual(_parse_versionexpr("2.3 3.2,  loose  = True", output), ("2.3 3.2", True, False))
+
+    def test_only_prerelease(self):
+        output = TestBufferConanOutput()
+        self.assertEqual(_parse_versionexpr("2.3, 3.2, include_prerelease=False", output), ("2.3 3.2", True, False))
+        self.assertEqual(_parse_versionexpr("2.3, 3.2, include_prerelease=True", output), ("2.3 3.2", True, True))
+
+    def test_both(self):
+        output = TestBufferConanOutput()
+        self.assertEqual(_parse_versionexpr("2.3, 3.2, loose=False, include_prerelease=True", output),
+                         ("2.3 3.2", False, True))
+        self.assertEqual(_parse_versionexpr("2.3, 3.2, include_prerelease=True, loose=False", output),
+                         ("2.3 3.2", False, True))
+
+    def test_invalid(self):
+        output = TestBufferConanOutput()
+        self.assertEqual(_parse_versionexpr("loose=False, include_prerelease=True", output), ("", False, True))
+        self.assertEqual(_parse_versionexpr("2.3, 3.2, unexpected=True", output),
+                         ("2.3 3.2 unexpected=True", True, False))
+        self.assertEqual(_parse_versionexpr("2.3, 3.2, loose=Other", output),
+                         ("2.3 3.2 loose=Other", True, False))
+        self.assertEqual(_parse_versionexpr("2.3, 3.2, ", output),
+                         ("2.3 3.2 ", True, False))
```

Comment:
I will add some invalid tests for things like:

```
"2.3, 3.2, 1.2.3"
"2.3 3.2; loose=True, include_prerelease=True"
"2.3\t3.2" <-- will tabs be supported?
"loose=True, 2.3 3.3"
```

Reply:
I tested it directly and it delivers no results back and unluckily no exception. This should be put into a new issue to follow up later (when something changed in the lib?).

Reply:
These are invalid inputs according to Conan docs, so they should behave exactly the same as a malformed version range. We cannot rely on `semver` doing the validation for us, they can change their behavior in future releases and they won't tell us.

Reply:
Look at https://github.com/conan-io/docs/issues/937